tests/puppeteer/pages/FO/FObasePage.js

```diff
@@ -0,0 +1,59 @@
+const CommonPage = require('../commonPage');
+
+module.exports = class Home extends CommonPage {
+  constructor(page) {
+    super(page);
+
+    // Selectors for home page
+    this.desktopLogo = '#_desktop_logo';
+    this.cartProductsCount = '#_desktop_cart span.cart-products-count';
+    this.userInfoLink = '#_desktop_user_info';
+    this.logoutLink = `${this.userInfoLink} .user-info a.logout`;
+    this.contactLink = '#contact-link';
+    this.categoryMenu = '#category-%ID > a';
+  }
+
+  /**
+   * go to the home page
+   */
+  async goToHomePage() {
+    await this.waitForSelectorAndClick(this.desktopLogo);
+  }
+
+  /**
+   * Filter by category
+   * @param categoryID, category id from the BO
+   */
+  async filterByCategory(categoryID) {
```

Comment:
We can filter categories from every FO page not only from home page.

Reply:
Not from CMS pages, from the Contact Us page, from the account page, login page... :smile:

Reply:
What we can do is create a CatalogPage which can contain these methods. Then you can have a HomePage which extends the CatalogPage ?

Reply:
Oops my bad you're right for these filters, even if they don't make really a lot of sense in CMS pages or Login page...
My only concern right now is for the name, since it's not really a filter but more of a link to a Category/Subcategory: `goToCategory` or `goToSubCategory` maybe ?

Reply:
Yes, it makes more sense to change the names of the functions ;)